api/scpca_portal/management/commands/load_metadata.py

```diff
@@ -0,0 +1,154 @@
+        )
+        parser.add_argument(
+            "--submitter-whitelist",
+            type=self.comma_separated_set,
+            default=common.SUBMITTER_WHITELIST,
+        )
+
+    def handle(self, *args, **kwargs):
+        self.load_metadata(**kwargs)
+
+    def comma_separated_set(self, raw_str: str) -> Set[str]:
+        return set(raw_str.split(","))
+
+    def can_process_project(
+        self, project_metadata: Dict[str, Any], submitter_whitelist: Set[str]
+    ) -> bool:
+        """
+        Validates that a project can be processed by assessing that:
+        - Input files exist for the project
+        - The project's pi is on the whitelist of acceptable submitters
+        """
+        project_path = common.INPUT_DATA_PATH / project_metadata["scpca_project_id"]
+        if project_path not in common.INPUT_DATA_PATH.iterdir():
+            logger.warning(
+                f"Metadata found for {project_metadata['scpca_project_id']},"
+                "but no s3 folder of that name exists."
+            )
+            return False
+
+        if project_metadata["pi_name"] not in submitter_whitelist:
+            logger.warning("Project submitter is not in the white list.")
+            return False
+
+        return True
+
+    def purge_project(
+        self,
+        project: Project,
+        *,
+        reload_existing: bool = False,
+        update_s3: bool = False,
+    ) -> bool:
+        """
+        Purges existing projects from the db so that they can be re-added when they are processed.
+        S3 is updated accordingly. Returns boolean as success status.
+        """
+        # Projects can only be intentionally purged.
+        # If the reload_existing flag is not set, then the project should not be procssed.
+        if not reload_existing:
+            logger.info(f"'{project}' already exists. Use --reload-existing to re-import.")
+            return False
+
+        # Purge existing projects so they can be re-added.
+        logger.info(f"Purging '{project}")
+        project.purge(delete_from_s3=update_s3)
+        return True
+
+    @staticmethod
+    def clean_up_input_data() -> None:
+        shutil.rmtree(common.INPUT_DATA_PATH, ignore_errors=True)
+
+    def load_metadata(
+        self,
+        input_bucket_name: str,
+        clean_up_input_data: bool,
+        reload_existing: bool,
+        scpca_project_id: str,
+        update_s3: bool,
+        submitter_whitelist: Set[str],
+        **kwargs,
+    ) -> None:
+        """Loads metadata from input metadata files on s3 and creates model objects in the db."""
+        # Prepare data input directory.
+        common.INPUT_DATA_PATH.mkdir(exist_ok=True, parents=True)
+
+        s3.download_input_metadata(input_bucket_name)
+
+        projects_metadata = metadata_file.load_projects_metadata(
+            Project.get_input_project_metadata_file_path(), scpca_project_id
+        )
```

Comment:
1. lets remove the first argument of this function and clean up if we can
2. lets force a kwarg on what is currently called `scpca_project_id` and make it `filter_on_project_id=scpca_project_id`